### PR TITLE
Disable discovery for spring cloud

### LINKED ISSUE #5162: Monitor Kubernetes permission issue
### Description

Similar to #5088, there is another warning indicating a lack of Kubernetes permissions. Both are regressions due to an overly broad dependency added in #5008.

### Steps to reproduce

Run the monitor in Kubernetes

### Additional context

```
2023-01-03T15:16:01.114-0600 ERROR boundedElastic-6 o.s.c.c.d.h.r.ReactiveDiscoveryClientHealthIndicator Error io.fabric8.kubernetes.client.KubernetesClientException:  Failure executing: GET at:
      https://10.8.0.1/api/v1/namespaces/previewnet/services. Message:
        Forbidden!Configured service account doesn't have access. Service account may have been revoked. services is forbidden: User "system:serviceaccount:previewnet:mirror-monitor" cannot list resource "services" in API group "" in the namespace "previewnet".
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:697)
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:676)
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:627)
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:566)
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:527)
  at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:510)
  at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:140)
  at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:422)
  at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:411)
  at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:91)
  at org.springframework.cloud.kubernetes.fabric8.discovery.KubernetesDiscoveryClient.getServices(KubernetesDiscoveryClient.java:307)
  at org.springframework.cloud.kubernetes.fabric8.discovery.KubernetesDiscoveryClient.getServices(KubernetesDiscoveryClient.java:303)
  at org.springframework.cloud.kubernetes.fabric8.discovery.reactive.KubernetesReactiveDiscoveryClient.lambda$getServices$1(KubernetesReactiveDiscoveryClient.java:60)
  at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:46)
  at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.run(FluxSubscribeOn.java:194)
  at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
  at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
  at io.micrometer.core.instrument.internal.TimedCallable.call(TimedCallable.java:51)
  at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
  at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
  at java.base/java.lang.Thread.run(Unknown Source)
```

### Hedera network

previewnet

### Version

0.70.2

### Operating system

None

### PR DESCRIPTION
**Pull request by jnels124**
Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

**Description**:

This PR disables spring cloud discovery as this mechanism isn't used in our environments

* Update dependencies for monitor to pull in only necessary functionality 
* Disable discovery for all executable modules

**Related issue(s)**:

Fixes #5162

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — APPROVED** [review]
Looks good

File: hedera-mirror-monitor/src/main/resources/application.yml [hunk 1]
```diff
@@ -61,6 +61,9 @@ server:
   port: 8082
   shutdown: graceful
 spring:
+  cloud:
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/src/main/resources/application.yml:64]: nit: Sort below `spring.application`

**Review by steven-sheehy — DISMISSED** [review]
LGTM

File: hedera-mirror-monitor/src/main/resources/application.yml (hunk 1 above)
Comment (jnels124) [reply hedera-mirror-monitor/src/main/resources/application.yml:66]: fixed.

**Review by steven-sheehy — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**